### PR TITLE
Removed legacy code and cleanup

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

See https://github.com/symfony/symfony/pull/32074, https://github.com/symfony/symfony/pull/26981

labels: `HttpFoundation`, `Form`, `Security`, `SecurityBundle`, `Validator`

### CODE REVIEW

**Comment by yceruto** [comment]
There is a `to be removed in 5.0` on this method:
https://github.com/symfony/symfony/blob/81241590520ce8338777072b6bdcc8600a00cbda/src/Symfony/Component/Form/FormConfigBuilder.php#L775-L777

What's the planned alternative for this case?

**Comment by Tobion** [comment]
The plan was to make the method private.

File: src/Symfony/Component/HttpFoundation/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ CHANGELOG
  * removed method `UploadedFile::getClientSize()` and the related constructor argument
  * made `Request::getSession()` throw if the session has not been set before
  * removed `Response::HTTP_RESERVED_FOR_WEBDAV_ADVANCED_COLLECTIONS_EXPIRED_PROPOSAL`
+ * passing a null url when instantiating a `egygfeggeewghgfewg` is not allowed
```
Comment (xabbuh) [inline src/Symfony/Component/HttpFoundation/CHANGELOG.md]: Typo in `FormError` :)

Comment (yceruto) [reply src/Symfony/Component/HttpFoundation/CHANGELOG.md]: Yap, that happens when you don't have a firewall between your daughter and the PC :) Fixed

**Comment by yceruto** [comment]
> The plan was to make the method private.

@Tobion but it's used out there (`ButtonBuilder`):
https://github.com/symfony/symfony/blob/d2711bfe0e206b571ef505c50bc1dbd4c1fec1ba/src/Symfony/Component/Form/ButtonBuilder.php#L64

**Review by nicolas-grekas — APPROVED** [review]

**Comment by fabpot** [comment]
Tests (low) are broken.

**Comment by yceruto** [comment]
Mmm, I'm not sure why, the broken tests are being eliminated here :thinking:

**Review by fabpot — APPROVED** [review]

**Comment by yceruto** [comment]
Tests green.

**Comment by fabpot** [comment]
Thank you @yceruto.

**Comment by Tobion** [comment]
@yceruto see #32177 and #32178

**Pull request merged**